### ISSUE TITLE
Please improve usage of language placeholders to make Kunena more friendly

### ISSUE THREAD

**Issue by joomlagate** [Kunena/Kunena-Forum#4922]
Let me take an example to make my point:

When someone replied a post on Kunena forum, Kunena will send out notification email to all subscribers to that topic. The subject of that email will be :

> A new reply has been posted on the xxxxxx

The "xxxxxx" string will be replaced with the real forum name in actural emails.

This looks fine, right?

Well, if you know a little about Chinese language, you will know that Chinese language uses DIFFERENT orders in this sentence. To us Chinese people, the following statement will be better :

> UUU at xxxxxx posted a new reply

Here, "UUU" stand for the username (you also can use "someone" as general name), and "xxxxxx" is still the forum name. The difference is, we put the forum name BEFORE the verb "posted".

It is not difficult to adjust the words orders in that sentence, if Kunena uses proper language variables (placeholders).  However, when I check the language difination of that sentence, it is:

`COM_KUNENA_POST_EMAIL_NOTIFICATION1="A new reply has been posted on the"`

Look, there is NO language variable (placeholder) for "forum name" in this language string. I think the correct one should be:

`COM_KUNENA_POST_EMAIL_NOTIFICATION1="A new reply has been posted on the %s1 by %s2"`

Where %s1 stand for "forum name" and %s2 stand for "username".

With such an improvement, both English language user and Chinese language user can be happy, because I will translate it to:

`COM_KUNENA_POST_EMAIL_NOTIFICATION1="%s2 at %s1 posted a new reply"`

Hope you will understand me and improve your usage about language variables.

Thank you.

**Comment by NoviceXue** [comment]
嘎嘎

**Comment by sshcli** [comment]
@baijianpeng 
Do you think that removing only "the" will solve the problem for the Chinese language ?

For example:
Now: A new reply has been posted on **the** board_title
After: A new reply has been posted on board_title

In my opinion, this will solve the problem for most of languages and I can create a PR for it in a few minutes.

In the PR I will remove "the" on these sentences:
A new post has been made to a category to which you have assigned as moderator on **the** 
A new reply has been posted on **the**
A new topic has been created on **the**

**Comment by joomlagate** [comment]
No, the word "the" will NEVER affect my translations, because it is STATIC string, not a variable.

In my above post, I tried to emphasize the "ordering" of those language variables, did you get my point?

Here is a simple example: like "user's real name", western people always write in this order:

First Name, Last Name (Family Name)

But Chinese language write in this order:

Family Name (without comma) First Name

Can you understand me now?

Some language variables in the language files of Kunena were arranged inappropriately so that I can not adjust their ordering in my translations.

For example, sometimes, there are two "%s" in the same language value, can you tell which is which? My suggestion is to use "%s1" and "%s2" , and "%s3" and so on, then we can recognize them easily and change their ordering according to different language custom.

Thank you.

**Comment by rich20** [comment]
Is this still with Kunena 5.0 ?   In my notification mail it shows the infos as followings:

> A new reply has been posted on the Forum
Message Subject : xxxxx
Category : xxxxx
Posted by : xxxxx
URL : xxxxx
Message: xxxxx

**Comment by sshcli** [comment]
>baijianpeng said: In my above post, I tried to emphasize the "ordering" of those language variables, did you get my point?

Yes, I did

> rich20 said: Is this still with Kunena 5.0 ?

Yes, the same applies for 5.0 and 5.1
